api/features/release_pipelines/core/models.py

```diff
@@ -72,7 +73,24 @@ def publish(self, published_by: FFAdminUser) -> None:
         self.published_at = timezone.now()
         self.published_by = published_by
         self.save()
-        self._create_pipeline_published_audit_log()
+
+    def unpublish(self, unpublished_by: FFAdminUser) -> None:
+        if self.published_at is None:
+            raise InvalidPipelineStateError("Pipeline is not published.")
+        self.published_at = None
+        self.published_by = None
+        self.save()
+
+    def clone(self) -> "ReleasePipeline":
```

Comment:
As it was raised by emyller in this [one](https://github.com/Flagsmith/flagsmith/commit/365ad46717cc4177936c5839eba272150e5a650c).

It doesn't feel right to perform the cloning in the models as deep down it's logic business and it ends up with:
- bloating the models
- having the logic scattered between the pipeline, stage, trigger and action models
while it indeed makes sense to have a service triggering the pipeline cloning and cascading down. 

I'd be in favor to use a service

Reply:
Do we plan to add the new name in the pipeline (as for segment?). Do we want to implement it now in the backend in that case?

Reply:
Okay, Done here: https://github.com/Flagsmith/flagsmith-private/pull/83